admin/admin-filters-sanitize.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Setup specific admin filters useful for sanitization.
+ *
+ * Extract from PLL_Admin_Filters to be able to use in a REST API context.
+ *
+ * @since 2.9
+ */
+class PLL_Admin_Filters_Sanitize {
+	/**
+	 * Current language used for the sanitization depending on the context.
+	 *
+	 * @var string $curlang
+	 */
+	public $curlang;
+
+	/**
+	 * Instance of PLL_Model
+	 *
+	 * @var PLL_Model
+	 */
+	protected $model;
+	
+	/**
+	 * Constructor: setups filters and actions
+	 *
+	 * @since 2.9
+	 *
+	 * @param object $polylang
+	 */
+	public function __construct( &$polylang ) {
+		$this->model = &$polylang->model;
+		$this->curlang = &$polylang->curlang;
+
+		// We need specific filters for some languages like German and Danish
+		$specific_locales = array( 'da_DK', 'de_DE', 'de_DE_formal', 'de_CH', 'de_CH_informal', 'ca', 'sr_RS', 'bs_BA' );
+		if ( array_intersect( $this->model->get_languages_list( array( 'fields' => 'locale' ) ), $specific_locales ) ) {
+			add_filter( 'sanitize_title', array( $this, 'sanitize_title' ), 10, 3 );
+			add_filter( 'sanitize_user', array( $this, 'sanitize_user' ), 10, 3 );
+		}
+	}
+
+	/**
+	 * Filters the locale according to the current language instead of the language
+	 * of the admin interface
+	 *
+	 * @since 2.0
+	 *
+	 * @param string $locale
+	 * @return string
+	 */
+	public function get_locale( $locale ) {
+		if ( isset( $_POST['post_lang_choice'] ) && $lang = $this->model->get_language( sanitize_key( $_POST['post_lang_choice'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			$locale = $lang->locale;
+		} elseif ( isset( $_POST['term_lang_choice'] ) && $lang = $this->model->get_language( sanitize_key( $_POST['term_lang_choice'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			$locale = $lang->locale;
+		} elseif ( isset( $_POST['inline_lang_choice'] ) && $lang = $this->model->get_language( sanitize_key( $_POST['inline_lang_choice'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			$locale = $lang->locale;
```

Comment:
I don't believe these cases will be relevant in a REST request context.

Reply:
PLL_Admin_Filters_Sanitize also concerns classical requests - ajax call for example when you modify the slug of the permalink in the classic editor context - not only REST ones

Reply:
Finally this code has been moved in PLL_Admin class because PLL_Filters_Sanitization instantiation is done in PLL_Admin::add_filters() method and need the locale and thus use this code to retrieve the right locale in a classic editor context